ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/manager/bounded/BoundedQueueTest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.eth.manager.bounded;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import tech.pegasys.pantheon.metrics.noop.NoOpMetricsSystem;
+
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+public class BoundedQueueTest {
+
+  @Test
+  public void offerShouldMakeARoomAndAcceptNewElementAtFullCapacity() {
```

Comment:
We should probably have some tests for cases where it doesn't need to remove an item.

Reply:
Test added.